classes/output/report_table.php

```diff
@@ -0,0 +1,178 @@
+        $this->set_attribute('class', 'generaltable generalbox');
+        $this->show_download_buttons_at(array(TABLE_P_BOTTOM));
+        $this->is_downloading($params->download, 'smartmedia-report');
+        $this->define_baseurl($url);
+        $this->define_columns(array('filename', 'type', 'width', 'duration', 'size', 'cost'));
+        $this->define_headers(array(
+            get_string('report:filename', 'local_smartmedia'),
+            get_string('report:type', 'local_smartmedia'),
+            get_string('report:resolution', 'local_smartmedia'),
+            get_string('report:duration', 'local_smartmedia'),
+            get_string('report:size', 'local_smartmedia'),
+            get_string('report:transcodecost', 'local_smartmedia')
+        ));
+        $this->currpage = isset($params->page) ? $params->page : 0;
+        $this->pagesize = isset($params->pagesize) ? $params->pagesize : 0;
+        $this->sortable(true);
+        $this->no_sorting('type');
+        $this->no_sorting('cost');
+        $this->set_sql(static::FIELDS, static::FROM, 'TRUE');
+
+    }
+
+    /**
+     * Get content for location column.
+     *
+     * @param \stdClass $row
+     *
+     * @return string html used to display the column field.
+     */
+    public function col_filename($row) {
+        return $this->format_text($row->filepath . $row->filename);
+    }
+
+    /**
+     * Get content for type column.
+     *
+     * @param \stdClass $row
+     *
+     * @return string html used to display the type field.
+     * @throws \moodle_exception
+     */
+    public function col_type($row) {
+        $metadata = json_decode($row->metadata);
+        return $this->format_text($metadata->formatname);
+    }
+
+    /**
+     * Get content for width column.
+     *
+     * @param \stdClass $row
+     *
+     * @return string html used to display the column field.
+     */
+    public function col_width($row) {
+        $resolution = $row->width . ' X ' . $row->height;
+        return $this->format_text($resolution);
+    }
+
+    /**
+     * Get content for duration column.
+     *
+     * @param \stdClass $row
+     *
+     * @return string html used to display the column field.
+     */
+    public function col_duration($row) {
+        return $this->format_text($row->duration);
+    }
+
+    /**
+     * Get content for size column.
+     *
+     * @param \stdClass $row
+     *
+     * @return string html used to display the column field.
+     */
+    public function col_size($row) {
+        $sizeinmb = $row->size / 1000000;
+        if (!$this->is_downloading()) {
+            $sizeinmb = round($sizeinmb, 2);
+        }
+        return $this->format_text($sizeinmb);
+    }
+
+    /**
+     * Get content for cost column.
+     *
+     * @param \stdClass $row
+     *
+     * @return string html used to display the column field.
+     */
+    public function col_cost($row) {
+        if ($row->width >= static::MIN_HD_WIDTH) {
```

Comment:
I know this will need to be all refactored once cost data is available, just be aware data is stored as seconds in metadata table.